File: position.py
import math

class position():
    def __init__(self, x, y):
        self.x = math.floor(x)
        self.y =  math.floor(y)
    def __add__(self, other):
        return position(self.x + other.x, self.y + other.y)
    def __sub__(self, other):
        return position(self.x - other.x, self.y - other.y)
    def min(p1, p2):
        return position(min(p1.x, p2.x), min(p1.y, p2.y))
    def max(p1, p2):
        return position(max(p1.x, p2.x), max(p1.y, p2.y))

File: src/position.c
#include "position.h"

#define max(a, b) (a >= b ? a : b)
#define min(a, b) (a <= b ? a : b)

position pos_min(position p1, position p2) {
  position output = {min(p1.x, p2.x), min(p1.y, p2.y)};
  return output;
}
position pos_max(position p1, position p2) {
  position output = {max(p1.x, p2.x), max(p1.y, p2.y)};
  return output;
}